### ISSUE TITLE
Consider to remove aria-haspopup

### ISSUE THREAD

**Issue by afercia** [moroshko/react-autowhatever#30]
Hello. Happy to see you're focusing in developing accessible components. Thanks for this 🙂 

I'm not sure [aria-haspopup](https://github.com/moroshko/react-autowhatever/blob/66f4456084930bc34a596dbe930da6a5b6b3e8a1/src/Autowhatever.js#L321) is used properly though.

`aria-haspopup` is a property, not a state. It's meant to indicate if an element has a popup that appears on some user activation, regardless of the popup visibility. In other words, when there's a popup, it should always be `true` even when the popup is not displayed. Worth nothing with ARIA 1.1 (still a Candidate Recommendation), things will change a bit and there will be [more values available](https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup). 

Additionally, unless I'm missing something, worth considering a combobox shouldn't have an `aria-haspopup` property at all. Unfortunately, the related section on the ARIA Authoring Practices is under some re-work https://www.w3.org/TR/wai-aria-practices/#combobox and they point to the related GitHub issue: https://github.com/w3c/aria-practices/issues/31

From there, you can refer to the sections that were temporarily removed (https://rawgit.com/w3c/aria-practices/master/aria-practices-DeletedSectionsArchive.html#combobox) and also to the section with general considerations about popups and drop-downs:

5.4 Popups and drop-downs
https://rawgit.com/w3c/aria-practices/master/aria-practices-DeletedSectionsArchive.html#relations_haspopup

where it's clearly stated:
> Combo boxes, or drop down lists, work differently. Controls with the role combobox must contain a list of items that can be opened, usually as a drop-down. Because this is intrinsic to the functionality of a combo box, it does not need to be explicitly indicated with aria-haspopup.

**Comment by moroshko** [comment]
@afercia Thanks a lot for such a thorough research!

Out of interest, I checked what other popular websites do, and found that:

* [Google](https://www.google.com/) - implements `aria-haspopup` with `true`/`false` values, like we do in `react-autowhatever`.
* [Bing](https://www.bing.com/) and [usa.gov](https://www.usa.gov/) - implement `aria-haspopup` which is always `true`, which supports your theory that it's a property, not a state.
* [Ebay](https://www.ebay.com/) - implements `aria-haspopup` which is always `false`, which seems weird.
* [Facebook](https://www.facebook.com/), [LinkedIn](https://www.linkedin.com/) and [Twitter](https://twitter.com/)- don't implement `aria-haspopup`, which supports your theory that it can be removed.

I wonder why Google decided to implement it this way. Any ideas?

**Comment by afercia** [comment]
> I wonder why Google decided to implement it this way. Any ideas?

@moroshko honestly I have no idea 🙂  My only (wild) guess is maybe some years ago the whole combobox pattern wasn't supported so well and adding an `aria-haspopup` was used as sort of fallback to communicate something appeared on the page. 

For sure, it's not a state: it's a property and its value shouldn't change.

Worth noting in the new ARIA 1.1 (still a Candidate Recommendation) things are a bit different and [aria-haspopup will have some new values](https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup) including "listbox". However, a combobox already has an implicit aria-haspopup property so it shouldn't be explicitly set, see:
https://www.w3.org/TR/wai-aria-1.1/#combobox
> Elements with the role combobox have an implicit aria-haspopup value of listbox.

**Comment by moroshko** [comment]
@afercia Cool, let's remove `aria-haspopup` then.
Mind creating a PR?

**Comment by sidoruk-sv** [comment]
@moroshko Hi!
Did I understand correctly that it should be just deleted?

**Comment by moroshko** [comment]
yeah

**Comment by sidoruk-sv** [comment]
@moroshko can you review #36 please

**Comment by moroshko** [comment]
Done.
